pkg/controller/nodeipam/ipam/multi_cidr_priority_queue.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ipam
+
+import (
+	"math"
+
+	cidrset "k8s.io/kubernetes/pkg/controller/nodeipam/ipam/multicidrset"
+)
+
+// A PriorityQueue implementation based on https://pkg.go.dev/container/heap#example-package-PriorityQueue
+
+// An PriorityQueueItem is something we manage in a priority queue.
+type PriorityQueueItem struct {
+	clusterCIDR *cidrset.ClusterCIDR
+	// labelMatchCnt is the first determinant of priority.
+	labelMatchCnt int
```

Comment:
labelMatchCount (try not to abbreviate by dropping vowels)

Reply:
Done